registrations/api.py

```diff
@@ -221,6 +227,44 @@ def send_message(self, request, pk=None, version=None):
             status=status.HTTP_200_OK,
         )
 
+    @action(
+        methods=["get"],
+        detail=True,
+        permission_classes=[CanAccessRegistration],
+        url_path=r"signups/export/(?P<file_format>xlsx)",
+    )
+    def signups_export(self, request, file_format=None, pk=None, version=None):
+        user = request.user
+        registration = self.get_object()
+
+        if not user.is_authenticated or not (
```

Comment:
Couldn't we move this checking logic to `CanAccessRegistration`?

Reply:
It would be a bit problematic as these checks only apply for the registrations' signups (a user requires additional permissions if they also want to view/manipulate the registrations' signups). Also, a registration is allowed to be fetched even by a non-authenticated user while signups are not.

Reply:
Ah but now I see what you meant: through `view.action` that would of course be possible. Another alternative could be to introduce an additional permission class for this case. I think I'll do some changes here

Reply:
Added a new permission class that is also used in the registration base serializer that has a similar check as this action endpoint.